### PR TITLE
tasks/postgres_upgrade.sh: shellcheck and related fixes

### LINKED ISSUE #356: shellcheck validation of tasks/postgres_upgrade.sh fails
simple scripting tweak required:
```
$ shellcheck gitlab/tasks/postgres_upgrade.sh

In gitlab/tasks/postgres_upgrade.sh line 12:
  if [ $DB_SIZE -lt $FREE ]; then
       ^------^ SC2086: Double quote to prevent globbing and word splitting.
                    ^---^ SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
  if [ "$DB_SIZE" -lt "$FREE" ]; then

For more information:
  https://www.shellcheck.net/wiki/SC2086 -- Double quote to prevent globbing ...
```

### PR DESCRIPTION
**Pull request by kenyon**
Some bash and command usage improvements. Fixes #356.

### CODE REVIEW

File: tasks/postgres_upgrade.sh [hunk 1]
```diff
@@ -4,12 +4,12 @@ echo 'Checking pgsql version'
 CMD=$(gitlab-psql --version)
 echo "version is ${CMD##* }"
 #9.2.18
-if [ "${CMD##* }" = "9.2.18" ]; then
+if [[ ${CMD##* } == "9.2.18" ]]; then
```
Comment (igalic) [inline tasks/postgres_upgrade.sh:7]: this makes the code bash only
are you sure that's what you want?

File: tasks/postgres_upgrade.sh [hunk 2]
```diff
@@ -4,12 +4,12 @@ echo 'Checking pgsql version'
...
   echo 'Version is below required for Gitlab 10+, checking...'
-  DB_SIZE=$(du -shk /var/opt/gitlab/postgresql/data | awk '{print $1}')
-  FREE=$(df -hk /var/opt/gitlab/postgresql/data/ | tail -1 | awk '{print $4}')
+  DB_SIZE=$(du -sk /var/opt/gitlab/postgresql/data | awk '{print $1}')
+  FREE=$(df -k /var/opt/gitlab/postgresql/data/ | tail -1 | awk '{print $4}')
   echo "Database size is: $DB_SIZE kb and freespace is $FREE kb"
-  if [ $DB_SIZE -lt $FREE ]; then
+  if (( DB_SIZE < FREE )); then
```
Comment (igalic) [inline tasks/postgres_upgrade.sh:12]: this makes the code bash only
are you sure that's what you want?

**Review by igalic — COMMENTED** [review]
i just noticed that the code already is bash-only, thanks to the shebang line

but if you revert (half) these changes, you could make it POSIX sh (maybe ;)

**Comment by kenyon** [comment]
Yes, that's what I want. :) Since it was already `#!/bin/bash`, might as well use the nice features of bash that make it a much safer programming language than plain sh.

**Review by igalic — APPROVED** [review]
🤷🏻‍♀️

**Pull request merged**